src/main/kotlin/uk/gov/dluhc/emsintegrationapi/database/entity/Address.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.dluhc.emsintegrationapi.database.entity
+
+import org.hibernate.annotations.CreationTimestamp
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import uk.gov.dluhc.emsintegrationapi.database.repository.UUIDCharType
+import uk.gov.dluhc.emsintegrationapi.database.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+
+@Entity
+class Address(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    @Type(type = UUIDCharType)
+    var id: UUID? = null,
+
+    @NotNull
+    val street: String,
+
+    var property: String? = null,
+    var locality: String? = null,
+    var town: String? = null,
+    var area: String? = null,
+    @NotNull
```

Comment:
use @field:NotNull on javax annotations.
I have found that these do not get carried to the java byte code unless you do this.

Will also need tests to validate this. We are doing this at the moment in OE (Shumail has a PR on it at the moment)

Reply:
Thanks Gurps, I have updated all the beans

Reply:
tests needed to prove validation is firing. See current OE PR for sample